event_routing_backends/processors/mixins/base_transformer.py

```diff
@@ -118,6 +118,15 @@ def extract_username_or_userid(self):
             username_or_id = self.get_data('data.username') or self.get_data('data.user_id')
         return username_or_id
 
+    def extract_sessionid(self):
+        """
+        Extracts sessionid of user from event either under context or data dictionaries
+
+        Returns:
+            str
+        """
+        return self.get_data('context.session') or self.get_data('data.session')
```

Comment:
Just curious, did you run into cases where the session data wasn't in the context? Do you think it would be possible to make this a required value?

Reply:
I assume this is some variation on the user's LMS session id that we're getting here? Asking because the xAPI spec seems to call for a something like a sub-session that is specific to the a video load. For example if you were to view the same video 3 times (say by leaving the page and coming back) in the same LMS session the spec seems to call for that to be 3 different "video sessions". At least that's my impression from what I'm reading here: https://liveaspankaj.gitbooks.io/xapi-video-profile/content/statement_data_model.html

Reply:
> Just curious, did you run into cases where the session data wasn't in the context?

Nope have not not found in data we can remove second part of that statement

> Do you think it would be possible to make this a required value?

You mean we should not transform event if session info is not present in the event?

Reply:
> I assume this is some variation on the user's LMS session id that we're getting here? Asking because the xAPI spec seems to call for a something like a sub-session that is specific to the a video load. For example if you were to view the same video 3 times (say by leaving the page and coming back) in the same LMS session the spec seems to call for that to be 3 different "video sessions". At least that's my impression from what I'm reading here: https://liveaspankaj.gitbooks.io/xapi-video-profile/content/statement_data_model.html

Yeah I agree. I think instead of using session-id extension of video profile we can create one in Open edX profile to reflect a user's session id.

Reply:
I'm also curious how this would work in a tracking log replay. I'm assuming that the session id baked into the log statement is the literal LMS Django session id. For privacy (and security) I think we should one-way hash that value as part of transformation.

Reply:
I would say we **must** hash it rather than **should.**  It should be consistent, but irreversible.

Reply:
@bmtcril session id before being pushed into log or any other backed is already hashed in [this](https://github.com/openedx/edx-platform/blob/197fef9c2b31de27a5a462bcdd2d8e7f167a492c/common/djangoapps/track/middleware.py#L211) middleware.

Reply:
Excellent, I learn something new every day. :)